### PR TITLE
Cherry-pick plugin permissions from Enterprise

### LINKED ISSUE #326: Cherry-pick plugin permissions from enterprise
Plugin permissions were implemented in an old version, cherrypick to DHC
https://deephaven.atlassian.net/browse/DH-11752
https://gitlab.eng.illumon.com/illumon/iris/merge_requests/6181/diffs

### PR DESCRIPTION
**Pull request by mofojed**
Plugin permissions were implemented in an old version, cherrypick to DHC
https://deephaven.atlassian.net/browse/DH-11752
https://gitlab.eng.illumon.com/illumon/iris/merge_requests/6181/diffs

Fixes #326

### CODE REVIEW

**Comment by mofojed** [comment]
Breaking change with the change in PropTypes in jsapi-shim.

File: packages/dashboard-core-plugins/src/prop-types/UIPropTypes.js [hunk 1]
```diff
@@ -33,12 +33,30 @@ const Link = PropTypes.shape({
...
+    isSuperUser: PropTypes.bool.isRequired,
+    isQueryViewOnly: PropTypes.bool.isRequired,
+    isNonInteractive: PropTypes.bool.isRequired,
+    canUsePanels: PropTypes.bool.isRequired,
+    canCreateDashboard: PropTypes.bool.isRequired,
+    canCreateCodeStudio: PropTypes.bool.isRequired,
+    canCreateQueryMonitor: PropTypes.bool.isRequired,
+    canCopy: PropTypes.bool.isRequired,
+    canDownloadCsv: PropTypes.bool.isRequired,
+  }),
```
Comment (vbabich) [inline packages/dashboard-core-plugins/src/prop-types/UIPropTypes.js:50]: I think `permissions` should be required
```suggestion
  }).isRequired,
```

File: packages/jsapi-shim/src/PropTypes.js [hunk 2]
```diff
@@ -69,7 +69,7 @@ const RollupConfig = PropTypes.shape({
   includeDescriptions: PropTypes.bool,
 });
 
-const User = PropTypes.shape({
+const UserInfo = PropTypes.shape({
```
Comment (vbabich) [inline packages/jsapi-shim/src/PropTypes.js:72]: I can't find how `UserInfo` is used. Shouldn't it be the same as `UIPropTypes.User`?

**Review by vbabich — CHANGES_REQUESTED** [review]

File: packages/jsapi-shim/src/PropTypes.js (hunk 2 above)
Comment (mofojed) [reply packages/jsapi-shim/src/PropTypes.js:72]: No, UserInfo is the object from the API, but it's only in Enterprise anyway so I'll take it out

**Review by vbabich — APPROVED** [review]

**Pull request merged**